OpenSim/Common/Storage.cpp

```diff
@@ -2299,16 +2299,21 @@ pad(int aPadSize)
 void Storage::
 smoothSpline(int aOrder,double aCutoffFrequency)
 {
+    int size = getSize();
     double dtmin = getMinTimeStep();
+    double avgDt = (_storage[size - 1].getTime() - _storage[0].getTime()) / size;
 
     if(dtmin<SimTK::Zero) {
         cout<<"Storage.SmoothSpline: storage cannot be resampled."<<endl;
         return;
     }
 
-    // RESAMPLE
-    dtmin = resample(dtmin,aOrder);
-    int size = getSize();
+    // RESAMPLE if the sampling interval is not uniform
+    if ((avgDt - dtmin) > SimTK::Eps) {
+        dtmin = resample(dtmin, aOrder);
+        size = getSize();
+    }
+
```

Comment:
+1. The only way for the minimum DT and the average DT to be the same is if you have uniform DT.